### PR TITLE
feat(caddy): php_server simplified directive

### PR DESCRIPTION
**Pull request by dunglas**
Introduce a new simplified `php_server` directive, which is very similar to [the `php_fastcgi` directive provided by Caddy](https://caddyserver.com/docs/caddyfile/directives/php_fastcgi).

This is now all you need to start a production-grade PHP server:

```Caddyfile
{
    frankenphp
    order php_server before reverse_proxy
}

localhost {
    encode zstd gzip
    php_server
    file_server
}
```

Would you mind checking if the code looks good to you and if it follows Caddy's best practices @francislavoie @mholt?

Replaces #92.

### CODE REVIEW

**Review by mholt — APPROVED** [review]
How did you know this is just what I always wanted for PHP :sob: and is precisely the reason I didn't originally name the `php_fastcgi` directive `php` -- I knew there had to be a better way to do PHP in the future! :100: :clap: 

approved

File: caddy/caddy.go [hunk 1]
```diff
@@ -283,12 +288,221 @@ func (f *FrankenPHPModule) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
...
+		routes = append(routes, redirRoute, rewriteRoute)
+	}
+
+	// route to actually reverse proxy requests to PHP files;
+	// match only requests that are for PHP files
+	pathList := []string{}
+	for _, ext := range extensions {
+		pathList = append(pathList, "*"+ext)
+	}
+	phpPatcherSet := caddy.ModuleMap{
```
Comment (mholt) [inline caddy/caddy.go:457]:
```suggestion
	phpMatcherSet := caddy.ModuleMap{
```

:man_shrugging: :question:

File: docs/config.md [hunk 2]
```diff
@@ -70,6 +70,40 @@ Alternatively, the short form of the `worker` directive can also be used:
...
+rewrite @indexFiles {http.matchers.file.relative}
+# FrankenPHP!
+@phpFiles path *.php
+php @phpFiles
+```
+
+The `php_server` and the `php` directives have the following options:
+
+```caddyfile
+php_server {
```
Comment (francislavoie) [inline docs/config.md:99]:
```suggestion
php_server [<matcher>] {
```

**Comment by mholt** [comment]
Thank you Francis, for being more thorough 😅 (It's been a stressful day here.)

**Pull request merged**